lib/puppet/provider/package/dpkg.rb

```diff
@@ -163,16 +163,7 @@ def unhold
     end
   end
 
-  def package_not_installed?(name)
-    if !name.nil? && !name.empty?
-      begin
-       dpkgquery("-W", "--showformat", self.class::DPKG_QUERY_FORMAT_STRING, name)
-      rescue Puppet::ExecutionFailure
-        # return true if exception is generated because package is not found
-        return true
-      end
-      return false
-    end
-    raise ArgumentError.new("Package name is nil or empty")
+  def package_not_installed?
+    query[:status] != "installed"
```

Comment:
Will the status be "installed" on non-en_US systems? Or does it return a localized string? But maybe that doesn't matter because Puppet sets the locale to "C" when executing the command?

Reply:
locale is not used by `dpkg-query` command on installed packages list:
```
root@wt1yi38jruqplkb:~# locale && dpkg-query -W --showformat '${Status} ${Package} ${Version}\n' postfix; 
LANG=fr_FR.utf8
...
unknown ok not-installed postfix 
```